rfcs/0000-central-station.md

```diff
@@ -0,0 +1,507 @@
+mode leverage and Swiftly's stated interest in consuming a feed like this; currently they only
+offer a proprietary API. If they don't get around to consuming a feed, CentralStation will instead
+use their API, tying us to Swiftly, but only at the CentralStation boundary, and not at Glides,
+Skate, Keolis levels.)
+
+CentralStation periodically produces the public GTFS VehiclePositions feed consumed by
+concentrate, by selecting its revenue vehicles and public fields. In the future, it may produce
+the Alerts and TripUpdates fields, too, using Swiftly and/or RTR and/or TransitClock as
+implementation details, and applying alerts and other business logic on top. But for now, we lean
+on the existing alert-applying logic in concentrate, and expect RTR and Swiftly to provide the
+TripUpdates to concentrate.
+
+If internal CTD apps need data not in the public feed, they can access it via CentralStation's
+API. For example, Glides will use CentralStation to access all the non-revenue vehicles, and Skate
+can get block and run information. (And operator info, like name? I don't know if PI info like
+that needs to live in CentralStation.)
+
+# Reference-level explanation
+
+[reference-level-explanation]: #reference-level-explanation
+
+Here are the effects of this proposal on the various pieces of our current realtime data
+ecosystem, starting with the idea of Service Changes, then all our existing apps, and finally
+CentralStation itself.
+
+## Service Changes
+
+A major component of this proposal, and a strong focus of CTD going forward, is the idea of
+"Service Changes". This is a [GTFS-rt proposal that has been discussed for 3.5
+years](https://github.com/google/transit/issues/113). Here is the [most recent version of the
+proposal](https://docs.google.com/document/d/1oPfQ6Xvui0g3xiy1pNiyu5cWFLhvrTVyvYsMVKGyGWM/edit#heading=h.mm6bblup6h8d).
+I refer to these in the RFC as a "ServiceChanges" file, but really the proposal is about new
+FeedEntity enum values, as well as extending the existing TripUpdates one. By convention, we
+produce `VehiclePositions` and `TripUpdates` files separately, though this is not required by the
+spec. We could similarly produce a `ServiceChanges` file composed of feed entities related to
+service changes.
+
+Swiftly has implemented some of it, in spirit, via its [Service
+Adjustments](https://www.goswift.ly/blog/introducing-service-adjustments) feature. Swiftly
+currently only supports creating these via a proprietary
+[API](https://swiftly-inc.stoplight.io/docs/service-adjustments/b3A6Mjg0MzkyMzY-create-adjustment)).
+
+The main thrust of it all is a way to update GTFS trips in real-time more comprehensively. Current
+TripUpdates can make predictions against existing trips, and can indicate that a train will skip a
+stop, but it can't indicate that the trip will go to a different stop, or provide a shape for that
+new route, or update properties of the trip like its headsign, limiting our ability to provide
```

Comment:
This sets out a large chunk of work for CentralStation, in additional to all the other features: drawing new shapes is a complicated interface!

Reply:
Even if we do build such an app, as part of CentralStation, whose job is it to input such information into it? Dispatchers? Inspectors? PIOs? How does it fit into their existing job duties? Would we get buy-in from leadership in the relevant parts of ops to have their employees putting work towards this? Some of this is for stuff that already gets put into an alert or something, but this is introducing a new level of detail and complexity.